gslib/tests/test_defacl.py

```diff
@@ -215,6 +226,70 @@ def testTooFewArgumentsFails(self):
+  def test_shim_translates_defacl_get(self):
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        mock_log_handler = self.RunCommand('defacl', ['get', 'gs://bucket'],
+                                           return_log_handler=True)
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(
+            ('Gcloud Storage Command: {} storage buckets describe'
+             ' --format=multi(defaultObjectAcl:format=json) gs://bucket').format(
+                 os.path.join('fake_dir', 'bin', 'gcloud')), info_lines)
+
+  @mock.patch.object(defacl.DefAclCommand, 'RunCommand', new=mock.Mock())
+  def test_shim_translates_set_defacl_file(self):
+    acl_string = '''[{
+      "entity": "project-owners-233984005590",
+      "etag": "CAE=",
+      "kind": "storage#objectAccessControl",
+      "projectTeam": {
+        "projectNumber": "233984005590",
+        "team": "owners"
+      },
+      "role": "OWNER"
+    }]'''
+    inpath = self.CreateTempFile(contents=acl_string.encode(UTF8))
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        mock_log_handler = self.RunCommand(
+            'defacl', ['set', inpath, 'gs://b1', 'gs://b2'],
+            return_log_handler=True)
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(
+            ('Gcloud Storage Command: {} storage buckets update'
+             ' --default-object-acl-file={} gs://b1 gs://b2').format(
+                 os.path.join('fake_dir', 'bin', 'gcloud'), inpath), info_lines)
+  
+  @mock.patch.object(defacl.DefAclCommand, 'RunCommand', new=mock.Mock())
+  def test_shim_translates_set_predefined_defacl(self):
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        mock_log_handler = self.RunCommand(
+            'defacl', ['set', 'predefined_acl', 'gs://b1', 'gs://b2'],
+            return_log_handler=True)
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(
+            ('Gcloud Storage Command: {} storage buckets update'
+             ' --predefined-default-object-acl={} gs://b1 gs://b2').format(
+                 os.path.join('fake_dir', 'bin', 'gcloud'), 'predefined_acl'), info_lines)
 
```

Comment:
Add a test for translating XML-style to JSON-style. Example: `public-read` to `publicRead` (see `test_acl.py`)

Reply:
Added translation test. And test_acl.py does not seem to have implemented or tested predefined_acl translation from xml to json?

Reply:
That would be because they were in my private branch 🙃

Made a PR: https://github.com/GoogleCloudPlatform/gsutil/pull/1653

Feel free to rebase on that, but there might be some changes while I solve the other parity bugs.